lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2646,6 +2657,65 @@ submitExternalTx tr nw tl sealedTx = do
         pure sealedTx
     pure txid
 
+votingEnabledInEra
+    :: Write.RecentEra era
+    -> Either ErrConstructTx ()
+votingEnabledInEra = \case
+    Write.RecentEraConway -> Right ()
+    _ -> Left ErrConstructTxVotingInWrongEra
+
+assertIsVoting
+    :: DBLayer IO s
+    -> Write.RecentEra era
+    -> ExceptT ErrConstructTx IO ()
+assertIsVoting DBLayer{..} = \case
+    Write.RecentEraConway -> do
+        voted <- liftIO $ atomically (alreadyVoted walletState)
+        if voted then
+            pure ()
+        else
+            throwE ErrConstructTxWithdrawalWithoutVoting
+    _ -> pure ()
+
+alreadyVoted
+    :: Functor stm
+    => DBVar stm (DeltaWalletState s)
+    -> stm Bool
+alreadyVoted walletState =
+    Dlgs.isVoting . view #delegations
+        <$> readDBVar walletState
+
+handleVotingWhenMissingInConway
+    :: Write.RecentEra era
+    -> DBLayer IO s
+    -> Bool
+    -> IO (Maybe VotingAction)
+handleVotingWhenMissingInConway era db delegationsPresent = do
+    areWeInConway <- case votingEnabledInEra era of
+        Left _ -> pure False
+        Right _ -> pure True
+    voting <- haveWeVoted db
+    stakingKeyRegistered <- isStakeKeyInDb db
+    if (areWeInConway && not voting && delegationsPresent) then
```

Comment:
```suggestion
    -> IO (Maybe VotingAction)
handleVotingWhenMissingInConway era db = do
    areWeInConway <- case votingEnabledInEra era of
        Left _ -> pure False
        Right _ -> pure True
    voting <- haveWeVoted db
    stakingKeyRegistered <- isStakeKeyInDb db
    if (areWeInConway && not voting) then
```

After removing the `handleVotingWhenMissingInConway` from `constructTransaction` and making it part of `selectCoinsForJoin`, the `delegationsPresent` parameter is now always `True` — and we can remove it. 🤓